packages/builders/src/components/ActionRow.ts

```diff
@@ -35,13 +41,14 @@ export class ActionRow<T extends ActionRowComponent = ActionRowComponent> implem
 	 * @param components The components to set this row to
 	 */
 	public setComponents(components: T[]) {
-		Reflect.set(this, 'components', [...components]);
+		this.components.slice(0, this.components.length);
+		this.components.push(...components);
```

Comment:
```suggestion
		this.components.splice(0, this.components.length, ...components);
```

Reply:
Also, just saying... this needs an unit test if nothing failed with this change, specially since `.slice` does not mutate the source array.

Reply:
As a side note, shouldn't we clone the array and do a direct assignment just like the old code did? This way we don't mutate input.

Reply:
> Also, just saying... this needs an unit test if nothing failed with this change, specially since `.slice` does not mutate the source array.

The source array gets overwritten in toJSON()

Reply:
I'm saying that `this.components.slice(0, this.components.length);` without return or assignment is basically a NOP.